services/tasks/service.go

```
@@ -331,12 +331,26 @@ func (s *Service) ListPids(ctx context.Context, r *api.ListPidsRequest) (*api.Li
 	if err != nil {
 		return nil, err
 	}
-	pids, err := t.Pids(ctx)
+	processList, err := t.Pids(ctx)
 	if err != nil {
 		return nil, errdefs.ToGRPC(err)
 	}
+	var processes []*task.ProcessInfo
+	for _, p := range processList {
+		var processInfo task.ProcessInfo
+		processInfo.Pid = p.Pid
+		if p.Info != nil {
+			a, err := typeurl.MarshalAny(p.Info)
+			if err != nil {
+				log.G(ctx).WithError(err).Errorf("Error marshalling ProcessInfo.Info for pid: %d\n", p.Pid)
```

Remove the log,

we can return the error instead:

 ```go
return nil, errors.Wrapf(err, "failed to marshal process %d info", p.Pid)
```